src/Uno.UI.Runtime.Skia.Gtk/UI/Xaml/Controls/TextBoxViewExtension.cs

```diff
@@ -61,10 +61,17 @@ public void StartEntry()
 				return;
 			}
 
+			// Selection is now handled by native control
+			_requestedSelectionStart = null;
+			_requestedSelectionLength = null;
+
 			_contentElement = textBox.ContentElement;
 			EnsureWidget(textBox);
 			var textInputLayer = GetWindowTextInputLayer();
-			textInputLayer.Put(_currentInputWidget!, 0, 0);
+			if (_currentInputWidget!.Parent != textInputLayer)
```

Comment:
Is `MemberNotNull` attribute available to use? Adding the following on `EnsureWidget` should allow us to remove the null suppressions.

```csharp
    [MemberNotNull(nameof(_currentInputWidget))]
```

Reply:
Good idea, that should work!

Reply:
Unfortunately we still compile against .NET Standard 2.0, so adding nullable attributes would require to recreate them in code or via a NuGet package. For simplicity sake, will probably keep the PR as is for now.